packages/multichain-network-controller/src/api/accounts-api.test.ts

```diff
@@ -0,0 +1,265 @@
+  invalidBitcoin: `${KnownCaipNamespace.Bip122}:1:invalid`,
+  unsupportedNamespace:
+    'cosmos:1:cosmos1hsk6jryyqjfhp5dhc55tc9jtckygx0eph6dd02',
+  customNamespace: `custom:1:${MOCK_ADDRESSES.evm}`,
+  testNamespace: `test:1:${MOCK_ADDRESSES.evm}`,
+};
+
+describe('toAllowedCaipAccountIds', () => {
+  type AccountType = {
+    type: EthAccountType | BtcAccountType | SolAccountType;
+    id: string;
+    options: Record<string, Json>;
+    methods: string[];
+    metadata: {
+      name: string;
+      importTime: number;
+      keyring: { type: string };
+    };
+    address: string;
+    scopes: `${string}:${string}`[];
+  };
+
+  const createMockAccount = (
+    address: string,
+    scopes: `${string}:${string}`[],
+    type: EthAccountType | BtcAccountType | SolAccountType,
+  ): AccountType => ({
+    address,
+    scopes,
+    type,
+    id: '1',
+    options: {},
+    methods: [],
+    metadata: {
+      name: 'Test Account',
+      importTime: Date.now(),
+      keyring: { type: 'test' },
+    },
+  });
+
+  it('formats account with EVM scopes', () => {
+    const account = createMockAccount(
+      MOCK_ADDRESSES.evm,
+      [EthScope.Eoa, EthScope.Mainnet, EthScope.Testnet],
+      EthAccountType.Eoa,
+    );
+
+    const result = toAllowedCaipAccountIds(account);
+    expect(result).toStrictEqual([
+      `${EthScope.Eoa}:${MOCK_ADDRESSES.evm}`,
+      `${EthScope.Mainnet}:${MOCK_ADDRESSES.evm}`,
+      `${EthScope.Testnet}:${MOCK_ADDRESSES.evm}`,
+    ]);
+  });
+
+  it('formats account with BTC scope', () => {
+    const account = createMockAccount(
+      MOCK_ADDRESSES.bitcoin,
+      [BtcScope.Mainnet],
+      BtcAccountType.P2wpkh,
+    );
+
+    const result = toAllowedCaipAccountIds(account);
+    expect(result).toStrictEqual([
+      `${BtcScope.Mainnet}:${MOCK_ADDRESSES.bitcoin}`,
+    ]);
+  });
+
+  it('formats account with Solana scope', () => {
+    const account = createMockAccount(
+      MOCK_ADDRESSES.solana,
+      [SolScope.Mainnet],
+      SolAccountType.DataAccount,
+    );
+
+    const result = toAllowedCaipAccountIds(account);
+    expect(result).toStrictEqual([
+      `${SolScope.Mainnet}:${MOCK_ADDRESSES.solana}`,
+    ]);
+  });
+
+  it('excludes unsupported scopes', () => {
+    const account = createMockAccount(
+      MOCK_ADDRESSES.evm,
+      [EthScope.Eoa, 'unsupported:123'],
+      EthAccountType.Eoa,
+    );
+
+    const result = toAllowedCaipAccountIds(account);
+    expect(result).toStrictEqual([`${EthScope.Eoa}:${MOCK_ADDRESSES.evm}`]);
+  });
+
+  it('returns empty array for account with no supported scopes', () => {
+    const account = createMockAccount(
+      MOCK_ADDRESSES.evm,
+      ['unsupported:123'],
+      EthAccountType.Eoa,
+    );
+
+    const result = toAllowedCaipAccountIds(account);
+    expect(result).toStrictEqual([]);
+  });
+});
+
+describe('toActiveNetworksByAddress', () => {
+  it('formats EVM network responses', () => {
```

Comment:
Used in other suggestions below:

```suggestion
describe('toActiveNetworksByAddress', () => {
  const SOLANA_MAINNET: CaipReference = '5eykt4UsFv8P8NJdTREpY1vzqKqZKvdp';

  it('formats EVM network responses', () => {
```